src/test/unit/services/sample/standalone_gqs_bad_test.cpp

```diff
@@ -6,47 +6,44 @@
 #include <test/test-models/good/services/test_gq2.hpp>
 #include <test/unit/services/instrumented_callbacks.hpp>
 #include <test/unit/util.hpp>
+#include <Eigen/Dense>
 #include <gtest/gtest.h>
 #include <iostream>
+#include <string>
+#include <vector>
 
 class ServicesStandaloneGQ2 : public testing::Test {
 public:
-  ServicesStandaloneGQ2()
-    : model(context, &model_log) {}
+  ServicesStandaloneGQ2() :
+    logger(logger_ss, logger_ss, logger_ss, logger_ss, logger_ss) {}
+
+  void SetUp() {
+    stan::io::empty_var_context context;
+    model = new stan_model(context);
+  }
+  
+  void TearDown() {
+    delete(model);
```

Comment:
Out of curiosity, why is this necessary?

Reply:
not sure what the scope of the question is - why is TearDown necessary?
https://www.ibm.com/developerworks/aix/library/au-googletestingframework.html
section on Test Fixtures:

> You can do deallocation of resources in TearDown or the destructor routine. However, if you want exception handling you must do it only in the TearDown code because throwing an exception from the destructor results in undefined behavior.

That said, it's here because I copied it from elsewhere in our unit test suite (advi?)

Reply:
I guess I'm not sure why we need to delete the model - does its destructor not get automatically called when a new one is created?

Reply:
If you're calling `new`, you should call `delete`. I understand that this is just test code, but there's no reason we should be leaking memory.

@seantalts, I believe the `SetUp()` method is setting up a model by calling `new` because we don't have default constructor for the model. So we're actually allocating memory for the object as we construct it.

Reply:
Ah okay, that makes sense. Sorry about that. I didn't know that trick and was assuming `new` was just a deprecated way of calling the default constructor, haha (we don't use it much!). My bad.

Reply:
nah... it's a good question. For our use on the autodiff stack, we redefine `operator new` so it grabs memory from our managed memory. Here, we're not using that `new`.

We don't do much with pointers for this reason... we need to keep track of every exit and make sure we've dealt with the memory properly. I haven't checked into C++11 initialization... there may be a way to hold onto the memory and then later initialize with the constructor without a default constructor. I'm not sure, though. (I thought I saw that while looking at the testing framework.)

Reply:
re-instated TearDown fns, per this discussion.